pyvista/utilities/fileio.py

```diff
@@ -182,10 +211,65 @@ def read_legacy(filename):
 
 
 def read(filename, attrs=None, force_ext=None, file_format=None):
-    """Read any VTK file.
-
-    It will figure out what reader to use then wrap the VTK object for
-    use in PyVista.
+    """Read any file type supported by ``vtk`` or ``meshio``.
+
+    Automatically determines the correct reader to use then wraps the
+    corresponding mesh as a pyvista object.  Attempts native ``vtk``
+    readers first then tries to use ``meshio``.
+
+    Supports the following formats:
+
+    Standard dataset readers:
+
+    * ``'.vtk'``
+    * ``'.pvtk'``
+    * ``'.vti'``
+    * ``'.pvti'``
+    * ``'.vtr'``
+    * ``'.pvtr'``
+    * ``'.vtu'``
+    * ``'.pvtu'``
+    * ``'.ply'``
+    * ``'.obj'``
+    * ``'.stl'``
+    * ``'.vtp'``
+    * ``'.vts'``
+    * ``'.vtm'``
+    * ``'.vtmb'``
+    * ``'.case'``
+
+    Image formats:
+
+    * ``'.bmp'``
+    * ``'.dem'``
+    * ``'.dcm'``
+    * ``'.img'``
+    * ``'.jpeg'``
+    * ``'.jpg'``
+    * ``'.mhd'``
+    * ``'.nrrd'``
+    * ``'.nhdr'``
+    * ``'.png'``
+    * ``'.pnm'``
+    * ``'.slc'``
+    * ``'.tiff'``
+    * ``'.tif'``
+
+    Other formats:
+
+    * ``'.byu'``
+    * ``'.g'``
+    * ``'.facet'``
+    * ``'.cas'``
+    * ``'.res'``
+    * ``'.foam'``
```

Comment:
I'm afraid some or a lot of these formats aren't fully supported. This all came up when @fbob asked about OpenFOAM files in slack. If you look at [their example](https://github.com/fbob/PyVista_OpenFOAM_Tutorial), it turns out that `pv.read(foam_file)` will automatically return only the first dataset in a single-element `MultiBlock`. They had to use the VTK reader with a call to `SetTimeValue` to get the dataset they needed (i.e. the one for the last time point).

It would be nice to be able to tell which formats are properly supported, but I suspect this will not be feasible to do, unless a given format is in 1:1 correspondence with .vtk. Still, a documented list of what _might_ work is still better than having to dig around the source code, so thanks for this.

Reply:
Perhaps adding a `.. note::` directive regarding certain file extensions would be helpful.  I'll add that for these file types.

Reply:
Also, I've created issue #1511.  I think the best bet here would be to create a new function to read openfoam files.  `pyvista.read` works when we don't need extra parameters, but it's best to have individual functions for different file types that have unique parameter combinations.

Reply:
Note rendered as:

![tmp](https://user-images.githubusercontent.com/11981631/125210627-4194b500-e25e-11eb-9a6c-06ba75208f3d.png)

Reply:
I'll cleanup the docstring...

Reply:
I was pointed to this thread in another PR.  The base `pyvista.read` does support all those files, but it reads them with the _default_ vtk reader settings.  The `attrs` parameter allows us to customize this behavior, but there aren't really good examples of doing this.  The user has to read the vtk documentation and it makes for a poor user experience so I don't think we should give examples.

There was a comment about comparing everything to the ".vtk" format. Although this is heavily prevalent, it is a legacy file format in vtk.  The newer XML format is much better IMO and has better reader support.  So, if there is a decision to have a "gold" standard format, I would prefer the XML format.  This is the sort of approach taken in #1536, where the ".vtk" format isn't yet supported.